### PR TITLE
Runnable löve2d

### PR DESCRIPTION
**Pull request by sornas**
- Change the makefile and add a `make run-main` that compiles main.sy into main.love and runs it.
- Make the lua-output runnable by löve by skipping printing a few statements.

### CODE REVIEW

File: sylt-compiler/src/lua.rs [hunk 1]
```diff
@@ -394,9 +394,9 @@ impl<'t> LuaCompiler<'t> {
         self.compiler.panic = false;
 
         match &statement.kind {
-            Use { .. } | Blob { .. } | IsCheck { .. } | EmptyStatement => {}
+            Use { .. } | Blob { .. } | IsCheck { .. } | EmptyStatement => return,
```
Comment (FredTheDino) [inline sylt-compiler/src/lua.rs:397]: This fixed all the empty statements?

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
-	@lua $@
+all: $(SYLT)
 
-clean:
-	rm -rf *.lua
+run-main: main.love
+	love .
+
+%.love: %.lua
+	cp $< $@
```
Comment (FredTheDino) [inline Makefile:11]: This should be a zip.
Something like this:
```
zip $< *.lua
```
But probably more.

I don't think you mean to have this rule here.

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
 
-%.lua: %.sy $(COMP)
-	@$(COMP) --lua $<
-	@echo "------ RUN ------"
-	@lua $@
+all: $(SYLT)
 
-clean:
-	rm -rf *.lua
+run-main: main.love
```
Comment (FredTheDino) [inline Makefile:7]: Shouldn't this depend on `main.lua`?

**Review by FredTheDino — CHANGES_REQUESTED** [review]
A bit strange...

File: sylt-compiler/src/lua.rs (hunk 1 above)
Comment (sornas) [reply sylt-compiler/src/lua.rs:397]: Well, it runs.

**Comment by sornas** [comment]
It appears I've forgotten how löve distributables work. V2 incoming.

```
::<sylt> make clean
rm -rf *.lua
rm -rf *.love
rm -rf build
::<sylt> make run
cargo build
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
mkdir -p build
target/debug/sylt --compile main.lua main.sy
mv main.lua build/main.lua
cp -r res/* build/
zip -9 -j main.love build/*
  adding: bloop.wav (deflated 72%)
  adding: coin.png (stored 0%)
  adding: example_game.sy (deflated 62%)
  adding: main.lua (deflated 79%)
  adding: sylt.png (deflated 0%)
  adding: sylt.svg (deflated 55%)
love main.love
::<sylt> make run MAIN=game.sy
cargo build
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
target/debug/sylt --compile game.lua game.sy
mv game.lua build/main.lua
cp -r res/* build/
zip -9 -j game.love build/*
  adding: bloop.wav (deflated 72%)
  adding: coin.png (stored 0%)
  adding: example_game.sy (deflated 62%)
  adding: main.lua (deflated 79%)
  adding: sylt.png (deflated 0%)
  adding: sylt.svg (deflated 55%)
love game.love
```

This way you can specify which .sy-file to compile. It defaults to `main.sy`.

File: Makefile (hunk 2 above)
Comment (sornas) [reply Makefile]: I don't think we depend on `*.lua` since all relevant .sy-files are compiled into the same file.

File: Makefile (hunk 3 above)
Comment (sornas) [reply Makefile]: We run the .love-file, not the .lua.

**Review by FredTheDino — APPROVED** [review]
LGTM

**Review by EMattfolk — APPROVED** [review]

**Pull request merged**